### ISSUE TITLE
wrapRoutes breaks FastifyInstance <> this binding

### ISSUE THREAD

**Issue by mjpowersjr** [autotelic/fastify-opentelemetry#28]
Fastify's default behavior is to bind the `this` variable to the Fastify Server instance. (see [Route Options](https://www.fastify.io/docs/latest/Routes/#routes-option)).

When using this plugin's `wrapRoutes` option, `this` is set to `undefined` when the route handler is invoked. Ideally, the original behavior could be maintained, while using wrapRoutes.

**Comment by jkirkpatrick24** [comment]
Hi @mjpowersjr !

Thanks for opening the issue, is there a minimum reproduction you could provide to help us fix this? Or better yet, would you like to contribute a PR?

**Comment by mjpowersjr** [comment]
Thanks for the quick response. I did take a look at the code, but TBH I'm a bit out of my depth on the opentelemetry side. This is a simple test that could be dropped along side your current tests.

```js
  test('should preserve this binding in handler', async ({ is, same, teardown }) => {
    class TestController {
        async handlerRequest(request, reply) {
            if (! this) {
                throw new Error('this is undefined');
            }
            return Object.keys(this)
        }
    } 


    // Note: fastify does not support binding 'this' with arrow functions based router handlers!
    const controller = new TestController();
    const fastify = setupTest({ 
        serviceName: 'test',
        wrapRoutes: false,
     }, controller.handlerRequest);

    const reply = await fastify.inject(injectArgs);
    is(reply.statusCode, 200);

    teardown(() => {
      fastify.close()
    })
  });
```

**Comment by HW13** [comment]
Thanks for the test snippet @mjpowersjr! I was able to track down the bug and have a fix in PR 👍

**Comment by HW13** [comment]
Fix is live in v0.11.2  🚀

**Comment by mjpowersjr** [comment]
Thanks you!
